src/main/webapp/mapsScript.js

```diff
@@ -44,26 +44,43 @@ window.initMap = function() {
   directionsRenderer.setMap(map);
   directionsRenderer.setPanel(document.getElementById('rightPanel'));
 
-  // Hard coded place IDs - will replace with user input
-  let timesSquareID = 'ChIJmQJIxlVYwokRLgeuocVOGVU';
-  let centralParkID = 'ChIJ4zGFAZpYwokRGUGph3Mf37k';
-  let worldTradeID = 'ChIJy7cGfBlawokR5l2e93hsoEA';
-  let empireStateID = 'ChIJtcaxrqlZwokRfwmmibzPsTU';
-  let hotelID = 'ChIJ68J3tfpYwokR2HaRoBcB4xg';
+  // Get locations from MapServlet and display directions on map.
+  displayRouteOnMap();
+}
+
+/*
+ * Gets locations from MapServlet with the tripKey parameter.
+ * Calls showDirections to show the directions with those locations.
+ */
+function displayRouteOnMap() {
+  const urlParams = new URLSearchParams(window.location.search);
+  const tripKey = urlParams.get('tripKey');
+  const tripKeyQuery = (tripKey != null && tripKey != '') ? '?tripKey=' + tripKey : '';
+  
+  fetch('/get-map' + tripKeyQuery).then(response => response.json()).then((locations) => {
+    showDirections(locations);
+  });
+}
+
+/* 
+ * Parses locations, generations DirectionsRequest with those locations,
+ * and displays on map with DirectionsRenderer.
+ * locations is a list of String addresses where the first element is the origin/destination
+ */
+function showDirections(locations) {
+  let origin = locations[0];
+  let waypts = [];
 
-  let waypts = [{location : {'placeId': worldTradeID}},
-                {location : {'placeId': empireStateID}},
-                {location : {'placeId': timesSquareID}},
-                {location : {'placeId': centralParkID}},
-              ];
+  for (let i = 1; i < locations.length; i++) {
+    waypts.push({ location : locations[i]});
+  }
```

Comment:
I chose to use a regular for loop since I need to exclude the first element of the list (which is the origin, not a waypoint). I don't know a clean way of using a foreach loop here, so leaving this as is

Reply:
Ok yes, that makes sense -- thanks!